transitgateway.tf

```diff
@@ -0,0 +1,26 @@
+# ------------------------------------------------------------------------------
+# Attach VPC to the Transit Gateway in the Shared Services account
+# (see https://github.com/cisagov/cool-sharedservices-networking).
+#
+# Note that this attachment will be automatically accepted as long
+# as the Transit Gateway was set up with:
+#  auto_accept_shared_attachments = "enable"
+# ------------------------------------------------------------------------------
+
+resource "aws_ec2_transit_gateway_vpc_attachment" "userservices" {
+  provider = aws.userservicesprovisionaccount
+
+  subnet_ids         = [for subnet in module.private.subnets : subnet.id]
+  tags               = var.tags
+  transit_gateway_id = local.transit_gateway_id
+  vpc_id             = aws_vpc.userservices.id
+}
+
+# Add a route to the User Services VPC in the TGW default route table
+resource "aws_ec2_transit_gateway_route" "userservices_route" {
+  provider = aws.sharedservicesprovisionaccount
+
+  destination_cidr_block         = aws_vpc.userservices.cidr_block
+  transit_gateway_attachment_id  = aws_ec2_transit_gateway_vpc_attachment.userservices.id
+  transit_gateway_route_table_id = local.transit_gateway_default_route_table_id
+}
```

Comment:
I don't think this block is necessary.  The route should be automatically generated and propagated to the default route table.  It probably makes sense to add a route to the _other_ (non-default) route tables, similar to what we do [here](https://github.com/cisagov/cool-sharedservices-networking/blob/develop/transitgateway.tf#L74-L84).
```suggestion
```

Reply:
Excellent catches!  Good thing you noticed that now- it would've taken me a while to troubleshoot this later when we first attempt sending traffic to the User Services VPC.

Please take a look at https://github.com/cisagov/cool-userservices-networking/pull/1/commits/d40c009445e7b12c53751a51d2c03c5f7cd7bcb3 and https://github.com/cisagov/cool-userservices-networking/pull/1/commits/6aec5ef6e2fb5d33a6a648f443a965d9709feb72 to ensure that I didn't misunderstand you.